packages/docusaurus-theme-common/src/contexts/doc.tsx

```diff
@@ -0,0 +1,73 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React, {useMemo, type ReactNode, useContext} from 'react';
+import {ReactContextError} from '../utils/reactUtils';
+import type {PropDocContent} from '@docusaurus/plugin-content-docs';
+
+/**
+ * The React context value returned by the `useDoc()` hook.
+ * It contains useful data related to the currently browsed doc.
+ */
+export type DocContextValue = {MDXComponent: () => JSX.Element} & Pick<
+  PropDocContent,
+  'metadata' | 'frontMatter' | 'toc' | 'assets' | 'contentTitle'
+>;
+
+const Context = React.createContext<DocContextValue | null>(null);
+
+/**
+ * Note: we don't use `PropDoc` as context value on purpose. Metadata is
+ * currently stored inside the MDX component, but we may want to change that in
+ * the future. This layer is a good opportunity to decouple storage from
+ * consuming API, potentially allowing us to provide metadata as both props and
+ * route context without duplicating the chunks in the future.
+ */
+function useContextValue(content: PropDocContent): DocContextValue {
+  return useMemo(
+    () => ({
+      MDXComponent: content,
```

Comment:
Do we want to expose this as context? Looks like TMI to me 😄

Reply:
That's part of the static data being available in the subtree, so not sure how it could be used, but why not?

Eventually we may move all this subtree context data to be made available at the plugin level? Using our secret `useRouteContext`?

https://github.com/facebook/docusaurus/issues/6885

You'd prefer to forward a `children` prop? 

I think both solutions are fine 🤷‍♂️ it's unlikely someone would want to render that component in a weird place like the Footer, so forwarding `children` could also work

Reply:
I was thinking this is useless in all cases to be exposed as a context, since only the main container would care about it and it doesn't really encourage any code reuse. (e.g. no one would be forwarding this to some subcomponent as props.)

Reply:
Up to you, I'm fine if you revert it to `children` if you think it's better, both look fine to me